gsy_framework/schema/avro_schemas/simulation_raw_data_v1.json

```diff
@@ -0,0 +1,562 @@
+                      "fields": [
+                        {"name": "market_fee", "type": "float"},
+                        {"name": "const_fee_rate", "type": "float"},
+                        {"name": "feed_in_tariff", "type": "float"},
+                        {"name": "market_maker_rate", "type": "float"},
+                        {
+                          "name": "bids",
+                          "type": {
+                            "name": "Bids",
+                            "type": "array",
+                            "items": ["Bid"]
+                          }
+                        },
+                        {
+                          "name": "offers",
+                          "type": {
+                            "name": "Offers",
+                            "type": "array",
+                            "items": ["Offer"]
+                          }
+                        },
+                        {
+                          "name": "trades",
+                          "type": {
+                            "name": "Trades",
+                            "type": "array",
+                            "items": ["Trade"]
+                          }
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Load",
+      "type": "record",
+      "fields": [
+        {"name": "market_fee", "type": "float"},
+        {"name": "load_profile_kWh", "type": "float"},
+        {"name": "total_energy_demanded_wh", "type": "float"},
+        {"name": "energy_requirement_kWh", "type": "float"},
+        {
+          "name": "bids",
+          "type": {
+            "name": "Bids",
+            "type": "array",
+            "items": ["Bid"]
+          }
+        },
+        {
+          "name": "offers",
+          "type": {
+            "name": "Offers",
+            "type": "array",
+            "items": []
+          }
+        },
+        {
+          "name": "trades",
+          "type": {
+            "name": "Trades",
+            "type": "array",
+            "items": ["Trade"]
+          }
+        }
+      ]
+    },
+    {
+      "name": "PV",
+      "type": "record",
+      "fields": [
+        {"name": "available_energy_kWh", "type": "float"},
+        {"name": "pv_production_kWh", "type": "float"},
+        {"name": "market_fee", "type": "float"},
+        {
+          "name": "bids",
+          "type": {
+            "name": "Bids",
+            "type": "array",
+            "items": []
+          }
+        },
+        {
+          "name": "offers",
+          "type": {
+            "name": "Offers",
+            "type": "array",
+            "items": ["Offer"]
+          }
+        },
+        {
+          "name": "trades",
+          "type": {
+            "name": "Trades",
+            "type": "array",
+            "items": ["Trade"]
+          }
+        }
```

Comment:
This is repeated a lot of times. Can't we do some definition and re-use it, like in JSONschema? An example is here:
https://github.com/gridsingularity/gsy-framework/blob/cd361fc1d9d0c920722930e7f117b2b82d020341/gsy_framework/schemas.py#L47

Reply:
This is almost doing the same thing (referencing to a previously defined schema), but I'll try to make it shorter in size (if possible).

Reply:
> This is almost doing the same thing (referencing to a previously defined schema) 

I see now. OK, I guess there is no possibility of improving this then. Thanks!